src/Classes/TradeQuery.lua

```diff
@@ -0,0 +1,897 @@
+currencyConversionTradeMap["Maven's Orb"] = "mavens-orb"
+currencyConversionTradeMap["Facetor's Lens"] = "facetors"
+currencyConversionTradeMap["Prime Regrading Lens"] = "prime-regrading-lens"
+currencyConversionTradeMap["Secondary Regrading Lens"] = "secondary-regrading-lens"
+currencyConversionTradeMap["Tempering Orb"] = "tempering-orb"
+currencyConversionTradeMap["Tailoring Orb"] = "tailoring-orb"
+currencyConversionTradeMap["Stacked Deck"] = "stacked-deck"
+currencyConversionTradeMap["Simple Sextant"] = "simple-sextant"
+currencyConversionTradeMap["Prime Sextant"] = "prime-sextant"
+currencyConversionTradeMap["Awakened Sextant"] = "awakened-sextant"
+currencyConversionTradeMap["Elevated Sextant"] = "elevated-sextant"
+currencyConversionTradeMap["Orb of Unmaking"] = "orb-of-unmaking"
+currencyConversionTradeMap["Blessing of Xoph"] = "blessing-xoph"
+currencyConversionTradeMap["Blessing of Tul"] = "blessing-tul"
+currencyConversionTradeMap["Blessing of Esh"] = "blessing-esh"
+currencyConversionTradeMap["Blessing of Uul-Netol"] = "blessing-uul-netol"
+currencyConversionTradeMap["Blessing of Chayula"] = "blessing-chayula"
+currencyConversionTradeMap["Veiled Chaos Orb"] = "veiled-chaos-orb"
+currencyConversionTradeMap["Enkindling Orb"] = "enkindling-orb"
+currencyConversionTradeMap["Instilling Orb"] = "instilling-orb"
+currencyConversionTradeMap["Sacred Orb"] = "sacred-orb"
+
+
+local TradeQueryClass = newClass("TradeQuery", function(self, itemsTab)
+    self.itemsTab = itemsTab
+	self.itemsTab.leagueDropList = { }
+    -- Note: Per each Check Price button click we do 2 search requests
+	--       Search is the most rate limiting behavior we need to track
+	-- SEARCH REQUEST RATE LIMIT DATA (as of Feb 2021)
+	--	 Up to  5 search requests in a 12 second window
+	--	 Up to 15 search requests in a 62 second window
+	--	 Up to 30 search requests in a 302 second window
+	self.totalPrice = { }
+	self.rate_short_window = { }
+	self.rate_short_time = 12
+	-- we reduce from 5 to 4 since we need 2 search slots for each request
+	self.rate_short_max_searches = 4
+	self.rate_medium_window = { }
+	self.rate_medium_time = 62
+	-- we reduce from 15 to 14 since we need 2 search slots for each request
+	self.rate_medium_max_searches = 14
+	self.rate_long_window = { }
+	self.rate_long_time = 302
+	-- we reduce from 30 to 29 since we need 2 search slots for each request
+	self.rate_long_max_searches = 29
+
+	-- FETCH REQUEST RATE LIMIT DATA (as of Feb 2021)
+	--	 Up to 12 fetch requests in a 6 second window
+	--	 Up to 16 fetch requests in a 14 second window
+	self.rate_short_fetch_window = { }
+	self.rate_short_fetch_time = 6
+	-- we reduce from 12 to 7 since we may want up to 5 fetch slots for each request
+	self.rate_short_max_fetches = 7
+	self.rate_long_fetch_window = { }
+	self.rate_long_fetch_time = 14
+	-- we reduce from 16 to 11 since we may want up to 5 fetch slots for each request
+	self.rate_long_max_fetches = 11
+
```

Comment:
These need to come from the GGG API response, not be hardcoded.  See https://www.pathofexile.com/developer/docs/index#ratelimits

Following those docs would also eliminate some of the helper methods as well.